src/common/io.py

```diff
@@ -146,31 +188,39 @@ def simulation_end(self):
         """
         pass
 
-    def set_options(self, options = None):
+    def set_options(self, options = None) -> None:
         """
         Options are the options dictionary provided to the simulation
         method.
         """
         self.options = options
 
-    def get_result(self):
+    def get_result(self) -> ResultReader:
         """
         Method for retrieving the result. This method should return a
-        result of an instance of ResultStorage or of an instance of a
-        subclass of ResultStorage.
+        result of an instance of ResultReader.
```

Comment:
Maybe I am missing something but currently `class ResultHandler` is not extending from any other class but the docstring says it should return an instance of `ResultReader`.

Reply:
Could we have ResultHandler also extend from abc.ABC just to make it more clear? Even though we dont have any abstract methods? I am not sure if its bad practice but just my initial thought.

Reply:
I also thought about it, but I'm not sure there is a point really. 

As you state, there are no real abstract methods, it is conceivable one just wants to do integrations points or diagnostics points. And without abstract methods, abstract base class has no meaning, doesn't even block instantiation of such classes.